server/sonar-server/src/main/java/org/sonar/server/permission/index/PermissionIndexerDao.java

```diff
@@ -111,6 +111,20 @@ public boolean isAllowAnyone() {
     "  project_authorization.qualifier as qualifier " +
     "FROM ( " +
 
+    // project is returned when no authorization
+    "      SELECT '" + RowKind.NONE + "' as kind," +
+    "      projects.uuid AS project, " +
+    "      projects.authorization_updated_at AS updated_at, " +
+    "      projects.qualifier AS qualifier, " +
+    "      NULL AS user_id, " +
+    "      NULL  AS group_id " +
+    "      FROM projects " +
+    "      WHERE " +
+    "        (projects.qualifier = 'TRK' or  projects.qualifier = 'VW') " +
+    "        AND projects.copy_component_uuid is NULL " +
+    "        {projectsCondition} " +
+    "      UNION " +
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "      UNION " 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1192)